example/src/main/java/io/netty/example/ocsp/OcspClientExample.java

```diff
@@ -0,0 +1,224 @@
+                @Override
+                public boolean verify(byte[] staple) throws Exception {
+                    if (staple == null) {
+                        throw new IllegalStateException("Server didn't provide an OCSP staple!");
+                    }
+
+                    OCSPResp response = new OCSPResp(staple);
+                    if (response.getStatus() != OCSPResponseStatus.SUCCESSFUL) {
+                        return false;
+                    }
+
+                    BasicOCSPResp basicResponse = (BasicOCSPResp) response.getResponseObject();
+                    SingleResp first = basicResponse.getResponses()[0];
+
+                    CertificateStatus status = first.getCertStatus();
+                    System.out.println(host);
+                    System.out.println("  Status: " + (status == CertificateStatus.GOOD ? "Good" : status));
+                    System.out.println("  This Update: " + first.getThisUpdate());
+                    System.out.println("  Next Update: " + first.getNextUpdate());
+
+                    if (first.getCertStatus() != CertificateStatus.GOOD) {
+                        return false;
+                    }
+
+                    return true;
+                }
+            });
+
+            EventLoopGroup group = new NioEventLoopGroup();
+            try {
+                FullHttpRequest request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "/");
+                request.headers().set(HttpHeaderNames.HOST, host);
+                request.headers().set(HttpHeaderNames.USER_AGENT, "netty-ocsp-example");
+
+                Promise<FullHttpResponse> promise = group.next().newPromise();
+
+                Bootstrap bootstrap = new Bootstrap()
+                        .channel(NioSocketChannel.class)
+                        .group(group)
+                        .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, 5 * 1000)
+                        .handler(newClientHandler(context, request, promise));
+
+                Channel channel = bootstrap.connect(host, 443)
+                        .syncUninterruptibly()
+                        .channel();
+
+                try {
+                    FullHttpResponse response = promise.get();
+                    ReferenceCountUtil.release(response);
+                } finally {
+                    channel.close();
+                }
+            } finally {
+                group.shutdownGracefully();
+            }
+        } finally {
+            context.release();
+        }
+    }
+
+    private static ChannelInitializer<Channel> newClientHandler(final ReferenceCountedOpenSslContext context,
+            final FullHttpRequest request, final Promise<FullHttpResponse> promise) {
+
+        return new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) throws Exception {
+                ReferenceCountedOpenSslEngine engine
```

Comment:
consider using `context.newHandler(ch.alloc())` instead (you can get the engine via `engine()`). This may produce some different behavior in the future (PR https://github.com/netty/netty/pull/6365) and I would like to make sure it still works at that time.